pkg/operator/controller/externaldns/pod.go

```diff
@@ -345,8 +335,9 @@ func (b *externalDNSContainerBuilder) fillGCPFields(container *corev1.Container)
 			container.Args = append(container.Args, fmt.Sprintf("--google-project=%s", *b.externalDNS.Spec.Provider.GCP.Project))
 		}
 	} else {
-		container.Args = append(container.Args, fmt.Sprintf("--google-project=%s", operatorv1alpha1.PlatformStatus.GCP.ProjectID))
+		container.Args = append(container.Args, fmt.Sprintf("--google-project=%s", b.platformStatus.GCP.ProjectID))
```

Comment:
This _should_ work because `platformStatus` _should_ always be set on a GCP cluster (`platformStatus` wasn't set on 4.1 clusters, but 4.1 couldn't be installed on GCP, so it should be impossible to have a GCP cluster that was upgraded from 4.1).  However, it is a pointer type, and the value is technically optional, so I'd still recommend having a nil check to be safe.

Reply:
Done